etc/tests/test_bootstrapper.py

```diff
@@ -0,0 +1,289 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import subprocess
+import nbformat
+import papermill
+import pytest
+import mock
+import sys
+import minio
+import os
+
+sys.path.append('etc/docker-scripts/')
+import bootstrapper
+
+
+@pytest.fixture
+def s3_setup(request):
+    bucket_name = "test-bucket"
+    cos_client = minio.Minio("127.0.0.1:9000",
+                             access_key="minioadmin",
+                             secret_key="minioadmin",
+                             secure=False)
+    cos_client.make_bucket(bucket_name)
+
+    def cleanup():
+        cleanup_files = cos_client.list_objects(bucket_name)
+        for file in cleanup_files:
+            cos_client.remove_object(bucket_name, file.object_name)
+        cos_client.remove_bucket(bucket_name)
+
+    request.addfinalizer(cleanup)
+
+    return cos_client
+
+
+def test_main_method(monkeypatch, s3_setup, tmpdir):
+    argument_dict = {'cos-endpoint': 'http://127.0.0.1:9000',
+                     'cos-bucket': 'test-bucket',
+                     'cos-directory': 'test-directory',
+                     'cos-dependencies-archive': 'test-archive.tgz',
+                     'notebook': 'etc/tests/resources/test-notebookA.ipynb',
+                     'inputs': 'test-file.txt',
+                     'outputs': 'test-file.txt'}
+    bucket_name = "test-bucket"
+    monkeypatch.setattr(bootstrapper, "parse_arguments", lambda x: argument_dict)
+    monkeypatch.setattr(bootstrapper, "package_install", lambda: True)
+    mocked_func = mock.Mock(return_value="default", side_effect=['test-archive.tgz',
+                                                                 'test-file.txt',
+                                                                 'test-notebookA-output.ipynb',
+                                                                 'test-notebookA.html',
+                                                                 'test-file.txt'])
+    monkeypatch.setattr(bootstrapper, "get_object_storage_filename", mocked_func)
+    monkeypatch.setenv("AWS_ACCESS_KEY_ID", "minioadmin")
+    monkeypatch.setenv("AWS_SECRET_ACCESS_KEY", "minioadmin")
+
+    s3_setup.fput_object(bucket_name=bucket_name,
+                         object_name="test-file.txt",
+                         file_path="README.md")
+    s3_setup.fput_object(bucket_name=bucket_name,
+                         object_name="test-archive.tgz",
+                         file_path="etc/tests/resources/test-archive.tgz")
+
+    with tmpdir.as_cwd():
+        bootstrapper.main()
+
```

Comment:
This confirms no exceptions where thrown, but that's about it.  We should add some validation that the test did what it was supposed to do.

Reply:
will check for existence of correct output files in both local and s3 file systems